tsl/src/bgw_policy/continuous_aggregate_api.c

```diff
@@ -484,6 +482,162 @@ parse_cagg_policy_config(const ContinuousAgg *cagg, Oid start_offset_type,
+		return POLICY_REFRESH_OFFSET_OVERLAP;
+
+	Hypertable *mat_ht = ts_hypertable_get_by_id(cagg->data.mat_hypertable_id);
+	const Dimension *dim = get_open_dimension_for_hypertable(mat_ht, true);
+
+	bool start_offset_isnull, end_offset_isnull;
+	int64 start_offset =
+		policy_refresh_cagg_get_refresh_start(cagg, dim, policy_config, &start_offset_isnull);
+	int64 end_offset = policy_refresh_cagg_get_refresh_end(dim, policy_config, &end_offset_isnull);
+
+	RangeBound lower = {
+		.val = Int64GetDatum(start_offset),
+		.infinite = start_offset_isnull,
+		.inclusive = true,
+		.lower = true,
+	};
+
+	RangeBound upper = {
+		.val = Int64GetDatum(end_offset),
+		.infinite = end_offset_isnull,
+		.inclusive = false,
+		.lower = false,
+	};
+
+	TypeCacheEntry *typcache = lookup_type_cache(INT8RANGEOID, TYPECACHE_RANGE_INFO);
+	if (typcache == NULL || typcache->rngelemtype == NULL)
+		elog(ERROR, "cache lookup failed");
+
+#if PG16_LT
+	RangeType *range = make_range(typcache, &lower, &upper, false);
+#else
+	RangeType *range = make_range(typcache, &lower, &upper, false, NULL);
+#endif
+
+	bool overlaps = false;
+	ListCell *lc;
+
+	elog(DEBUG1,
+		 "start_offset: " INT64_FORMAT ", end_offset: " INT64_FORMAT,
+		 start_offset,
+		 end_offset);
+
+	/* We need to go through all jobs in order to determine if there is an existing job with the
+	 * exact same offsets */
+	foreach (lc, jobs)
+	{
+		BgwJob *job = (BgwJob *) lfirst(lc);
+
+		if (existing_job_id == job->fd.id)
+		{
+			continue;
+		}
+
+		bool start_offset_job_isnull, end_offset_job_isnull;
+		int64 start_offset_job = policy_refresh_cagg_get_refresh_start(cagg,
+																	   dim,
+																	   job->fd.config,
+																	   &start_offset_job_isnull);
+
+		int64 end_offset_job =
+			policy_refresh_cagg_get_refresh_end(dim, job->fd.config, &end_offset_job_isnull);
+
+		RangeBound lower_job = {
+			.val = Int64GetDatum(start_offset_job),
+			.infinite = start_offset_job_isnull,
+			.inclusive = true,
+			.lower = true,
+		};
+
+		RangeBound upper_job = {
+			.val = Int64GetDatum(end_offset_job),
+			.infinite = end_offset_job_isnull,
+			.inclusive = false,
+			.lower = false,
+		};
+
+#if PG16_LT
+		RangeType *range_job = make_range(typcache, &lower_job, &upper_job, false);
+#else
+		RangeType *range_job = make_range(typcache, &lower_job, &upper_job, false, NULL);
+#endif
+
+		elog(DEBUG1,
+			 "start_offset_job: " INT64_FORMAT ", end_offset_job: " INT64_FORMAT,
+			 start_offset_job,
+			 end_offset_job);
+
+		/* Check if exact same job exists, in which case throw an error or notice depending on
+		 * `if_not_exists` */
+		if (start_offset == start_offset_job && end_offset == end_offset_job)
+		{
+			/* If all arguments are the same, do nothing */
+			return POLICY_REFRESH_OFFSET_OVERLAP_EQUAL;
+		}
+		else if (range_overlaps_internal(typcache, range_job, range))
+		{
+			overlaps = true;
+		}
+	}
+
+	if (overlaps)
+	{
+		return POLICY_REFRESH_OFFSET_OVERLAP_NONE;
+	}
```

Comment:
Is this correct? Shouldn't it be the opposite?

```suggestion
	if (!overlaps)
	{
		return POLICY_REFRESH_OFFSET_OVERLAP_NONE;
	}
```

Reply:
I see that the enum is using a different naming than I expected.